geopandas/explore.py

```diff
@@ -490,16 +490,33 @@ def _style_color(x):
 
             style_function = _style_color
         else:  # assign new column
+            # allow for matplotlib color abbreviations, e.g. "r" for "red"
+            def plt_color(c):
+                def hex_c(c):
+                    return (
+                        colors.to_hex(c)
+                        if isinstance(c, str)
+                        and colors.is_color_like(c)
+                        and not c[0] == "#"
+                        else c
+                    )
+
+                return (
+                    [hex_c(c_) for c_ in c]
+                    if pd.api.types.is_list_like(c)
+                    else hex_c(c)
```

Comment:
Why not remove `hex_c` and simply use `[colors.to_hex(c_) for c_ in colors.to_rgba_array(c)]`, as  `to_hex` already handles well either color abbreviations, full names, RGB(A) or hexadecimal representations? The `matplotlib.colors` functions probably provide a flexibility in inputs that's hard to beat. Also I think passing an RGB(A) would break the current implementation, whereas if you use `[colors.to_hex(rgba) for rgba in colors.to_rgba_array(c)]`, I think simply checking whether the list has a single element a posteriori could work.

Reply:
there are two reasons I have coded it the way it is.  1) `color` parameter can be a string or list like.  list comprehension is only used when it's list like.  2) do not use `to_hex()` on colors that are strings that start with `#` so that unit tests still work.  hence `hex_c()` is used in both contexts of string and list like and also caters for not *splatting* colors that are already a hexadecimal formatted string

Reply:
The thing is, a color coded as RGB(A) (like for black: `(0, 0, 0, 1)`) is list-like, so your implementation breaks if a user passes such a color, which they can do in `matplotlib`, so some will try it with `explore` too. Regarding 2), I don't understand your point about "splatting colors", but what I know is that `colors.to_rgba_array` handles every kind of input we consider here well ((list/array of) named colors, abbreviated names, RGB(A), including hex strings), and outputs it to an RGB(A) array on which it's easy to work on with `to_hex`. Also, the solution I propose may sound far from optimal, but doing this color normalisation is completely negligible time-wise compared to generating the HTML with leaflet anyway, so better doing this part in a non optimal but robust way, in my opinion.